ext/openssl/xp_ssl.c

```diff
@@ -2143,6 +2144,9 @@ static ssize_t php_openssl_sockop_io(int read, php_stream *stream, char *buf, si
 		if (nr_bytes > 0) {
 			php_stream_notify_progress_increment(PHP_STREAM_CONTEXT(stream), nr_bytes, 0);
 		}
+		if (old_eof != stream->eof) {
+			php_stream_notify_completed(PHP_STREAM_CONTEXT(stream));
+		}
```

Comment:
Why is this not done in `_php_stream_fill_read_buffer` resp. `_php_stream_write_buffer`?

Reply:
Right, that's better indeed. Thx.
I had put it in the places where the progress notification already was, because that seemed like it would fit together, especially since notifications are only supported on http(s) & ftp streams. But it makes sense to make it more general, I'll change it.